Add steps for ensuring pre-commit hooks are installed

dragid10 opened the pull request:
Added steps for running tests and pre-commit linters/formatters.

## Additions:
- Add more explicit steps for installing dependencies (codespace and local)
- Add steps for explicitly installing pre-commit hooks (codespace and local)
- Add explicit steps for committing and pushing to GitHub
- Added screenshots for instructions added 

## Changes:

- Make command for starting jekyll server run in background
  - Since `bundle exec` is greedy and steals the terminal from the user, we send it to the background so that the user can then run other commands easily

Closes https://github.com/BlackPythonDevs/blackpythondevs.github.io/issues/351

dragid10 committed:
Add steps for ensuring pre-commit hooks are installed

dragid10 committed:
Add .idea folder to gitignore

dragid10 committed:
Add steps for running tests and linters/formatters

dragid10 committed:
Merge remote-tracking branch 'upstream/gh-pages' into gh-pages

kjaymiller commented on README.md:
```diff
@@ -12,6 +12,114 @@ This repo is the [GitHub Pages](https://pages.github.com/) website, built in Jek
+Follow these steps and note these guidelines to begin contributing:
+
+1. First step is to set up the local development environment.
+1. Bug fixes are always welcome. Start by reviewing the [list of bugs](https://github.com/BlackPythonDevs/blackpythondevs.github.io/issues).
```
```suggestion
2. Bug fixes are always welcome. Start by reviewing the [list of bugs](https://github.com/BlackPythonDevs/blackpythondevs.github.io/issues).
```

kjaymiller commented on README.md:
```diff
@@ -12,6 +12,114 @@ This repo is the [GitHub Pages](https://pages.github.com/) website, built in Jek
+
+1. First step is to set up the local development environment.
+1. Bug fixes are always welcome. Start by reviewing the [list of bugs](https://github.com/BlackPythonDevs/blackpythondevs.github.io/issues).
+1. A good way to easily start contributing is to pick and work on a [good first issue](https://github.com/BlackPythonDevs/blackpythondevs.github.io/labels/good%20first%20issue). We try to make these issues as clear as possible and provide basic info on how the code should be changed, and if something is unclear feel free to ask for more information on the issue.
```
```suggestion
3. A good way to easily start contributing is to pick and work on a [good first issue](https://github.com/BlackPythonDevs/blackpythondevs.github.io/labels/good%20first%20issue). We try to make these issues as clear as possible and provide basic info on how the code should be changed, and if something is unclear feel free to ask for more information on the issue.
```

kjaymiller commented on README.md:
Great work on breaking this information down. One thing I would like to do is move these steps into CONTRIBUTING.md.

dragid10 committed:
Retain trace logs on failure

Allows us to better debug why the tests just occasionally fail

dragid10 committed:
Add tracing to playwright tests

dragid10 committed:
Run tracing action regardless of test status

dragid10 committed:
Fix folder where test results will upload

dragid10 committed:
Use proper tests dir

dragid10 committed:
Actually turn on tracing and upload the resulting failed results

dragid10 committed:
Use jekyll serve detach command

I somehow missed that the command existed

dragid10 committed:
Add macos, jetbranins and vscode to gitignore

dragid10 committed:
Remove sleep fixture

Playwright has a built-in "slowmo" feature that would be easiest to use.
https://playwright.dev/python/docs/test-runners#cli-arguments

dragid10 committed:
Use standard python comparison operator

The `to_equal` doesn't seem to exist in playwright for python. There is a NodeJS equivalent, but there is no supported GenericAssertion equivalent for python

https://playwright.dev/docs/api/class-genericassertions

dragid10 committed:
Add explanation to test assertion

dragid10 committed:
Fix linting issues

dragid10 committed:
Merge pull request #1 from dragid10/fix-playwright-tests

dragid10 committed:
Merge branch 'gh-pages' into gh-pages

dragid10 committed:
Pull upstream changes

dragid10 committed:
Use white theme for images

dragid10 committed:
Remake changes in CONTRIBUTING file

dragid10 committed:
Remove unneeded test-results folder

dragid10 committed:
Revert .gitignore

dragid10 committed:
Revert to base readme

dragid10 committed:
Merge branch 'upstream-gh-pages' into gh-pages

dragid10 committed:
Use white theme to stay consistent

dragid10 committed:
Increase GH action job to 30 mins

dragid10 committed:
Run linter to fix linting errors

dragid10 commented:
Okay lots of commits later (we should definitely do a squash and merge for this one), and it is finally ready for review! @kjaymiller

kjaymiller commented on .github/workflows/playwright.yml:
@dragid10 - should it take 30 minutes? I don't think I've seen it take that long ever

dragid10 replied:
I doubt it would, but as I was testing some changes on the GH actions, I found out that it has a very long (or no) timeout. So if something funky starts happening, then I'd like to kill the actions job in a reasonable amount of time

kjaymiller replied:
I would consider like maybe 10 minutes then. I really don't imaging it getting much longer than that (of course until it does)

kjaymiller replied:
but that can be added in a new issue. Great work on this one!
